From 7c4838dfa520bc564ac940f86de49b7b511a34c6 Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Wed, 8 Aug 2012 21:00:01 +0800
Subject: ENGR00219837-2 HDMI ARM2: Add HDMI_SDMA support for RIGEL TO1.1

Replace mx6q_version() check with hdmi_SDMA_check() to add support for
HDMI_SDMA in RIGEL TO1.1.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 5d9600d..5ed2715 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -53,6 +53,7 @@
 #include <linux/mfd/max17135.h>
 #include <sound/pcm.h>
 #include <linux/mxc_asrc.h>
+#include <linux/mfd/mxc-hdmi-core.h>
 
 
 #include <mach/common.h>
@@ -1412,7 +1413,7 @@ static void hdmi_init(int ipu_id, int disp_id)
 	mxc_iomux_set_gpr_register(3, 2, 2, hdmi_mux_setting);
 
 	/* Set HDMI event as SDMA event2 while Chip version later than TO1.2 */
-	if ((mx6q_revision() > IMX_CHIP_REVISION_1_1))
+	if (hdmi_SDMA_check())
 		mxc_iomux_set_gpr_register(0, 0, 1, 1);
 }
 
-- 
1.7.9.5

